packages/headless/src/database/mongodb/user-auth/index.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { UserEntity } from 'src/entity/user';
+import {
+  CreateUserDto,
+  UserEntityResponse,
+} from 'src/modules/user-auth/dto/user.dto';
+import { IUserAuthDB } from 'src/modules/user-auth/repository/user.db.interface';
+import { UserModel } from './user.model';
+
+@Injectable()
+export class UserAuthDB implements IUserAuthDB {
+  async save(body: CreateUserDto): Promise<UserEntityResponse> {
+    const isExist = await UserModel.findOne({ phone: body.phone }).lean();
+    if (isExist) {
```

Comment:
Please don't use here if-else that break The Single Responsibility Principle. Because databases do have not other dependencies.